lib/src/obsoleted/analyzer_plugin/analyzer_plugin.dart

```diff
@@ -187,13 +187,18 @@ class MetricsAnalyzerPlugin extends ServerPlugin {
     ResolvedUnitResult analysisResult,
   ) {
     try {
-      if (analysisResult.unit != null) {
+      final path = analysisResult.path;
+      if (analysisResult.unit != null &&
+          path != null &&
+          (driver.analysisContext?.contextRoot.isAnalyzed(path) ?? false)) {
```

Comment:
It seems that we have null `analysisContext` and this won't work (that was the problem with `optionsFilePath`).